obsei/source/pandas_source.py

```diff
@@ -0,0 +1,103 @@
+import logging
+from obsei import configuration
+from typing import List, Optional, Dict, Any
+
+from pandas.core.frame import DataFrame
+from obsei.source.base_source import BaseSource, BaseSourceConfig
+from obsei.payload import TextPayload
+from obsei.misc.utils import (
+    DATETIME_STRING_PATTERN,
+    DEFAULT_LOOKUP_PERIOD,
+    convert_utc_time,
+)
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+
+class PandasSourceConfig(BaseSourceConfig):
+    TYPE: str = "PandasScrapper"
+    target: str = None
+    target_df: Optional[pd.DataFrame] = None
+    input_type: str = None
+    rows: Optional[int] = None
+    columns: list = None
+
+    def __init__(self, **data: Any):
+        super().__init__(**data)
+
+        # self.target = None
+        # self.input_type: str = None
+        # self.rows: Optional[int] = None
+        # self.columns: Optional[list] = None
+
+
+class PandasSourceScrapper(BaseSource):
+    NAME: Optional[str] = "PandasSourceScrapper"
+
+    def lookup(  # type: ignore[override]
+        self, config: PandasSourceConfig, **kwargs
+    ) -> List[TextPayload]:
+        source_responses: List[TextPayload] = []
+
+        # Get data from state
```

Comment:
I think pandas dataframe can't have any state information. But we can keep last execution time information.